framework/base/Event.php

```diff
@@ -110,23 +137,40 @@ public static function on($class, $name, $handler, $data = null, $append = true)
     public static function off($class, $name, $handler = null)
     {
         $class = ltrim($class, '\\');
-        if (empty(self::$_events[$name][$class])) {
+        if (empty(self::$_events[$name][$class]) && empty(self::$_eventWildcards[$name][$class])) {
             return false;
         }
         if ($handler === null) {
             unset(self::$_events[$name][$class]);
```

Comment:
Could be written as `unset(self::$_events[$name][$class], self::$_eventWildcards[$name][$class])`.

Reply:
I would better not do it: arguments are long, such refactoring will create a long statement, which is hard to comprehend.